jsonschema-core.md

```diff
@@ -2146,540 +2119,48 @@ a fragment in order to identify the specific schema object.
-  property `unspecified-prop`.
-- The value at `/foo/foo-prop` evaluated at
-  `/properties/foo/allOf/1/properties/foo-prop` by following the path
-  `/properties/foo/allOf/1/properties/foo-prop` by the `const` keyword is not
-  the constant value 1.
-- The value at `/bar/bar-prop` evaluated at `/$defs/bar/properties/bar-prop` by
-  following the path `/properties/bar/$ref/properties/bar-prop` by the `type`
-  keyword is not a number.[^16][^17]
-
-[^16]: `minimum` doesn't produce an error because it only operates on instances
-that are numbers.
-
-[^17]: Note that the error message wording as depicted in the examples below is
-not a requirement of this specification. Implementations SHOULD craft error
-messages tailored for their audience or provide a templating mechanism that
-allows their users to craft their own messages.
-
-The passing instance will produce the following annotations:
-
-- The keyword `title` evaluated at `` by following the path `` will produce
-  `"root"`.
-- The keyword `properties` evaluated at `` by following the path `` will produce
-  `["foo", "bar"]`.
-- The keyword `title` evaluated at `/properties/foo` by following the path
-  `/properties/foo` will produce `"foo-title"`.
-- The keyword `properties` evaluated at `/properties/foo/allOf/1` by following
-  the path `/properties/foo/allOf/1` will produce `["foo-prop"]`.
-- The keyword `additionalProperties` evaluated at `/properties/foo/allOf/1` by
-  following the path `/properties/foo/allOf/1` will produce
-  `["unspecified-prop"]`.
-- The keyword `title` evaluated at `/properties/foo/allOf/1/properties/foo-prop`
-  by following the path `/properties/foo/allOf/1/properties/foo-prop` will
-  produce `"foo-prop-title"`.
-- The keyword `title` evaluated at `/$defs/bar` by following the path
-  `/properties/bar/$ref` will produce `"bar-title"`.
-- The keyword `properties` evaluated at `/$defs/bar` by following the path
-  `/properties/var/$ref` will produce `["bar-prop"]`.
-- The keyword `title` evaluated at `/$defs/bar/properties/bar-prop` by following
-  the path `/properties/bar/$ref/properties/bar-prop` will produce
-  `"bar-prop-title"`.
-
-#### Flag
-
-In the simplest case, merely the boolean result for the "valid" valid property
-needs to be fulfilled. For this format, all other information is explicitly
-omitted.
-
-```json
-{
-  "valid": false
-}
-```
+### Instance Location
 
-Because no errors or annotations are returned with this format, it is
-RECOMMENDED that implementations use short-circuiting logic to return failure or
-success as soon as the outcome can be determined. For example, if an `anyOf`
-keyword contains five subschemas, and the second one passes, there is no need to
-check the other three. The logic can simply return with success.
+The instance location is the location of the JSON value within the root instance
+being validated. The value MUST be expressed as a JSON Pointer.
```

Comment:
I have a use case in mind related to a future hyper-schema-ish implementation where instances would have IRIs and output results may include more than one instance. It would be nice if the instance location was defined in a way that could support such a case. If this was defined as an IRI instead of just a JSON Pointer, current applications could use a fragment-only IRI fragment and it would leave open the possibility of future uses that might want to use the a full IRI.

I could introduce a new field type when I eventually have the need, but that would mean it wouldn't be compatible with existing tooling, which would be a bummer.

Another option is to introduce a new field later that is just the absolute IRI of the instance and that would work in conjunction with this keyword. That would probably at least mostly work, but the naming would be weird. A better name might be `instancePointer`. Then if we introduce a new field to be used for the instance IRI, the naming is more consistent with `schemaLocation` which is also an IRI.

If you want to discuss this separately, that's fine, but I wanted to mention once more that this being just a pointer is potentially limiting for novel applications of JSON Schema.

Reply:
> and output results may include more than one instance.

This isn't covered by JSON Schema anywhere.  Throughout the spec, an evaluation takes place between a single (root) schema and a single instance.  A secondary instance means a secondary evaluation.

Secondly, while an instance may _have_ an identifier, the specification does not mention or make use of them.

It doesn't make sense for output to allow for multiple instances or instance identifiers when they're not mentioned elsewhere in the spec.  (And, no, I'm not interested in adding that.  Such things are extension behaviors.)

Reply:
> This isn't covered by JSON Schema anywhere.

In our charter we say we want to enable and support the use of JSON Schema in ways that it wasn't specifically designed for. I see this as falling under that umbrella. We can't predict every way JSON Schema might be used, but when we have someone (me) saying they have an idea for novel way to apply JSON Schema, we shouldn't be knowingly throwing up roadblocks for them. We should be trying to clear the roadblocks.

> an evaluation takes place between a single (root) schema and a single instance

Hyper-schema adds links to plain JSON via a schema. The variation of hyper-schema I have in mind adds references (`$ref`) to plain JSON via a schema. So, just like you have a single (root) schema, you still have a single (root) instance. Just like you can end up in a different schema through a reference, you can end up in a different JSON document through a reference.

> Such things are extension behaviors.

Of course it's an extension, but there's no reason an implementation that supports this extension shouldn't be able to use a standard output format and tools that make use of that format. The output for evaluations using this extension can be this same as a standard evaluation. There's no difference other than that there could be more than one instance involved.

---

Do whatever you want. I think it's closed-minded, but I'm tired of every little thing turning into a fight with you and this one isn't worth it.

Reply:
> In our charter we say we want to enable and support the use of JSON Schema in ways that it wasn't specifically designed for.

I hear you, but the spec is still written with validation and annotation of a single JSON instance as its primary function.  The terms defined herein need to reflect that.

As mentioned [above](https://github.com/json-schema-org/json-schema-spec/pull/1429#discussion_r1322173829), you're welcome to define additional terms that make sense in the context of evaluating multiple instances.